Test failing against pip main branch

https://github.com/jazzband/pip-tools/actions/runs/4475375698/jobs/7864734569

```python
    def copy_install_requirement(
        template: InstallRequirement, **extra_kwargs: Any
    ) -> InstallRequirement:
        """Make a copy of a template ``InstallRequirement`` with extra kwargs."""
        # Prepare install requirement kwargs.
        kwargs = {
            "comes_from": template.comes_from,
            "editable": template.editable,
            "link": template.link,
            "markers": template.markers,
            "use_pep517": template.use_pep517,
            "isolated": template.isolated,
>           "install_options": template.install_options,
            "global_options": template.global_options,
            "hash_options": template.hash_options,
            "constraint": template.constraint,
            "extras": template.extras,
            "user_supplied": template.user_supplied,
        }
E       AttributeError: 'InstallRequirement' object has no attribute 'install_options'

```

--- a/piptools/utils.py
+++ b/piptools/utils.py
@@ -21,6 +21,7 @@ from pip._vendor.packaging.utils import canonicalize_name
 from pip._vendor.packaging.version import Version
 from pip._vendor.pkg_resources import Distribution, Requirement, get_distribution
 
+from piptools._compat import PIP_VERSION
 from piptools.subprocess_utils import run_python_snippet
 
 _KT = TypeVar("_KT")
@@ -456,7 +457,6 @@ def copy_install_requirement(
         "markers": template.markers,
         "use_pep517": template.use_pep517,
         "isolated": template.isolated,
-        "install_options": template.install_options,
         "global_options": template.global_options,
         "hash_options": template.hash_options,
         "constraint": template.constraint,
@@ -464,6 +464,9 @@ def copy_install_requirement(
         "user_supplied": template.user_supplied,
     }
     kwargs.update(extra_kwargs)
+
+    if PIP_VERSION[:2] <= (23, 0):
+        kwargs["install_options"] = template.install_options
 
     # Original link does not belong to install requirements constructor,
     # pop it now to update later.

